### PR TITLE
In some cases, the lseek may causes offset less than 0 to make read/write error

### PR DESCRIPTION
**Pull request by zhangduan-inspur**
In some cases, the lseek may causes offset less than 0 to make read/write error

### CODE REVIEW

File: src/client/Client.cc [hunk 1]
```diff
@@ -8558,8 +8559,15 @@ loff_t Client::_lseek(Fh *f, loff_t offset, int whence)
...
+  if (f->pos < 0){
+    f->pos = old_pos;
+    r = -EINVAL;
+  }else{
+    r = f->pos;
+  }
+
   ldout(cct, 3) << "_lseek(" << f << ", " << offset << ", " << whence << ") = " << f->pos << dendl;
-  return f->pos;
+  return r;
```
Comment (branch-predictor) [inline src/client/Client.cc]: That's not a correct way to handle it. _lseek should cap the offset to range 0..size and return capped offset.

Comment (zhangduan-inspur) [reply src/client/Client.cc]: As I understand it, the input offset  can be negative.
The return value -1 is indicating the error, so do you mean that the return value can only be -1 or a valid positive value?
Thanks

Comment (branch-predictor) [reply src/client/Client.cc]: In that particular case, there's nothing wrong in seeking file by negative offset, past file beginning. In that case, it should just rewind to 0 and not signal an error condition.

Comment (zhangduan-inspur) [reply src/client/Client.cc]: Thanks for the advice. I'll update the code and pull request again。

Comment (branch-predictor) [reply src/client/Client.cc]: You can reopen this PR and just amend your commit.

**Comment by zhangduan-inspur** [comment]
I create a new PR（#17604）for this，please review for me，thanks。

**Pull request closed without merging**